File: Day14/minReorder.cpp
int minReorder(int n, vector<vector<int>>& connections) {
    sort(connections.begin(), connections.end());
    set<int> path;
    path.insert(0);
    vector<vector<int>> left;
    int count = 0;
    for(auto const& i : connections){
        if(path.find(i[1]) != path.end())
            path.insert(i[0]);
        else if(path.find(i[0]) != path.end()){
            count++;
            path.insert(i[1]);
        }
        else
            left.push_back(i);
        
    }
    for(auto const& i : left) {
        if(path.find(i[1]) != path.end())
            path.insert(i[0]);
        else if(path.find(i[0]) != path.end()){
            count++;
            path.insert(i[1]);
        }
    }
    return count;
}

File: Day14/minReorder.java
public int minReorder(int n, int[][] connections) {
    Arrays.sort(connections, (a,b) -> a[1]-b[1]);
    Set<Integer> path = new HashSet<>();
    path.add(0);
    List<int[]> left = new ArrayList<>();
    int cnt = 0;
    for(int[] i : connections) {
        if(path.contains(i[1]))
            path.add(i[0]);
        else if(path.contains(i[0])){
            cnt++;
            path.add(i[1]);
        } else {
            left.add(i);
        }
    }
    for(int[] i : left) {
        if(path.contains(i[1]))
            path.add(i[0]);
        else if(path.contains(i[0])){
            cnt++;
            path.add(i[1]);
        }
    }
    return cnt;
}